src/target/linux/writemem.rs

```diff
@@ -0,0 +1,133 @@
+use nix::unistd::Pid;
+use std::{marker::PhantomData, mem};
+
+/// Allows to write data to different locations in debuggee's memory as a single operation.
+/// This implementation can select different strategies for different memory pages.
+pub struct WriteMemory<'a> {
+    pid: Pid,
+    write_ops: Vec<WriteOp>,
+    _marker: PhantomData<&'a mut ()>,
```

Comment:
```suggestion
    _marker: PhantomData<&'a ()>,
```

`WriteMemory` only requires immutable references, not mutable ones. This changes the variance of the lifetime parameter to allow a bit more code.

Reply:
Good catch, thanks!